### PR TITLE
Better document language overriding capabilities

### PR DESCRIPTION
**Pull request by pschriner**
You can override labels used in <f:translate> outside of an Extbase context if an extensionName is provided or the extension name can be inferred from the language key (LLL:EXT:extensionkey)

### CODE REVIEW

File: Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst [hunk 1]
```diff
@@ -56,7 +56,9 @@ Overriding translations in non-Extbase plugins does not work out of the box.
 
 ..  attention::
     Setting :php:`_LOCAL_LANG` has no effect if the ViewHelper
-    :html:`<f:translate>` is used outside of the Extbase context.
+    :html:`<f:translate>` is used outside of the Extbase context
+    and the `extensionName` attribute is not set and the key used
+    does not follow the LLL:EXT:extensionkey syntax.
```
Comment (garvinhicking) [inline Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:61]:
```suggestion
    does not follow the `LLL:EXT:extensionkey` syntax.
```

File: Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst [hunk 2]
```diff
@@ -56,7 +56,9 @@ Overriding translations in non-Extbase plugins does not work out of the box.
 
 ..  attention::
     Setting :php:`_LOCAL_LANG` has no effect if the ViewHelper
-    :html:`<f:translate>` is used outside of the Extbase context.
+    :html:`<f:translate>` is used outside of the Extbase context
+    and the `extensionName` attribute is not set and the key used
```
Comment (garvinhicking) [inline Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:60]:
```suggestion
    and the :html:`extensionName` ViewHelper attribute is not set and the key used
```

File: Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst [hunk 3]
```diff
@@ -56,7 +56,9 @@ Overriding translations in non-Extbase plugins does not work out of the box.
 
 ..  attention::
     Setting :php:`_LOCAL_LANG` has no effect if the ViewHelper
-    :html:`<f:translate>` is used outside of the Extbase context.
+    :html:`<f:translate>` is used outside of the Extbase context
```
Comment (garvinhicking) [inline Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:59]: I wonder if we should negate this whole section, like:

```
Setting :php:`_LOCAL_LANG` can only have an effect if:
* :html:`<f:translate>` is used inside Extbase context (not a Fluid StandaloneView),
* OR the ViewHelper's :html:`extensionName` attribute is set to the proper extension key,
* OR the ViewHelper's :html:`key` attribute uses the `LLL:EXT:extensionkey/Path/To/locallang.xlf:languagekey` syntax.
```

Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:59]: I still don't see how this is supposed to work outside of Extbase. Which part of code would apply the `_LOCAL_LANG` settings then? If I am for example in the context of a FLUIDTEMPLATE, how would this work?

Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:59]: So we should then say extbase and fluid standalone?

Comment (garvinhicking) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:59]: Hhhhm... maybe?! I don't know. If "Non-Extbase" is the same as "Fluid Standalone", then yes. I don't know if there is another scenario to use _LOCAL_LANG that is something else than "Fluid Standalone" but not "Extbase"...?

Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:59]: Also this does not fit with line 55 here.

Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:59]: Do you have an example where this is actually working? Or some code reference?

Comment (pschriner) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:59]: Minimal example:

page = PAGE
page.10 = FLUIDTEMPLATE
page.10.template = TEXT
page.10.template.value (

  Hello <f:translate key="onlineDocumentation" extensionName="backend" />
  
  Hello <f:translate key="LLL:EXT:backend/Resources/Private/locallang.xlf:onlineDocumentation" />
  
)

Comment (pschriner) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:59]: maybe even the example is worth documenting. I didn't dig to the very end why you have to prefix the extension with tx_, but not using tx_ will not work

Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst:59]: Wow, very interesting! Yes I think it would be worth documenting this!

Comment (garvinhicking) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst]: I like the current approach, that's very helpful! Thanks so much!

**Review by garvinhicking — APPROVED** [review]

**Review by linawolf — APPROVED** [review]

File: Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst (hunk 3 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/Localization/TypoScript.rst]: I like it too, thanks a lot

**Comment by linawolf** [comment]
Thanks a lot, merged it and backported it manually as the automatic backport workflow was failing

**Pull request merged**